Barotrauma/BarotraumaShared/SharedSource/Prefabs/PrefabCollection.cs

```diff
@@ -174,12 +176,34 @@ public void AddNodesAndInheritors(IEnumerable<Identifier> ids)
             public void InvokeCallbacks()
             {
                 HashSet<Node> uncheckedNodes = IdToNode.Values.ToHashSet();
+                // We no longer need to worry about parent self references, but maybe this will still catch other issues???
```

Comment:
Cycles are still a problem in cases like:
- `A` inherits from `B`
- `B` inherits from `C`
- `C` inherits from `A`